### ISSUE TITLE
Missing Cargo.lock in nightly, unable to build stdlib

### ISSUE THREAD

**Issue by dunric** [rust-lang/rustup#3984]
### Verification

- [X] I searched for recent similar issues at https://github.com/rust-lang/rustup/issues?q=is%3Aissue+is%3Aopen%2Cclosed and found no duplicates.
- [X] I am on the latest version of Rustup according to https://github.com/rust-lang/rustup/tags and am still able to reproduce my issue.

### Problem

Recent _nightly_ toolchain package for *x86_64-unknown-linux-gnu* platform miss in its *rust-src* component `Cargo.lock` file, which rust compiler expects when rebuild of standard library is required.

### Steps

1. Make sure toolchain  `nightly-x86_64-unknown-linux-gnu` is installed with its `rust-src` component
2. Attempt to compile a cargo project with `build-std` option like
  `cargo +nightly build -Z build-std=std,panic_abort --target=x86_64-unknown-linux-gnu -r`
3. Compilation fails with
> error: "/home/whatever/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/Cargo.lock" does not exist, unable to build with the standard library, try:
        rustup component add rust-src --toolchain nightly-x86_64-unknown-linux-gnu

`rust-src` for nightly is indeed installed and subdirectory _~/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust_ contains _bin/_ and _library/_ from nightly, only `Cargo.lock` is not there.

### Rustup version

```console
rustup 1.27.1 (54dd3d00f 2024-04-24)
```

### Installed toolchains

```console
stable-x86_64-unknown-linux-gnu (default)
nightly-x86_64-unknown-linux-gnu
```

### OS version

```console
Linux x86_64 (Arch Linux)
```

**Comment by rami3l** [comment]
@dunric Thanks for filing this issue! I'm not sure if this is a Rustup issue, but it might indicate a faulty installation caused by something like #988. For the moment being, I suggest uninstalling and reinstalling the toolchain in question.

**Comment by ChrisDenton** [comment]
This is a Cargo issue and is fixed by https://github.com/rust-lang/cargo/pull/14370

**Comment by dunric** [comment]
> This is a Cargo issue and is fixed by [rust-lang/cargo#14370](https://github.com/rust-lang/cargo/pull/14370)

Aha, thanks for the quick reply. Good to know there is already PR which fixes this issue.
